ports/cef/window.rs

```diff
@@ -442,20 +442,22 @@ impl WindowMethods for Window {
         }
     }
 
-    fn set_page_url(&self, url: ServoUrl) {
-        // it seems to be the case that load start is always called
-        // IMMEDIATELY before address change, so just stick it here
-        on_load_start(self);
+    fn history_changed(&self, history: HistoryEntries) {
         let browser = self.cef_browser.borrow();
         let browser = match *browser {
             None => return,
             Some(ref browser) => browser,
         };
+
+        let current = history.current;
+        let can_go_back = current > 0;
+        let can_go_forward = current < history.entries.len() - 1;
```

Comment:
These are the forward/back settings for the session history of the frame that just navigated, not the joint session history.